src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -235,7 +235,7 @@ public void removeAll() {
             }
         }
 
-        return tokens.iterator();
+        return tokens;
```

Comment:
From the code it appears that 'nulls' may be in the Iterable<TokenCacheItem> returned. Because we create the collection as ArrayList<TokenCacheItem> tokens = new ArrayList<TokenCacheItem>(results.values().size()); If decrypt fails, we will not add the tokenCacheItem.